### ISSUE TITLE
Tuist Dump Regression from 4.32.0 

### ISSUE THREAD

**Issue by danibachar** [tuist/tuist#7143]
### What happened?

When running tuist dump in parallel, NIO errors can be introduced, I suspect its due to [this](https://github.com/tuist/tuist/commit/757df2db71794280e5085ea9289bd67c2aa77c7a#diff-9e5385c2d611fd977c011ecf6238035ec3cb55ce4e929b6ebb96eec41c711ecfR12) change I made utilizing the `CachedManifestLoader` for the dump command.

If indeed that change is in charge of it - would the Tuist team be open to revert it given we have the protection check [here](https://github.com/tuist/tuist/commit/affac241587d77c64d87550e6031fa55633efd62#diff-70cba27f5cddb0850785b77ded2a515d37cf0b771fe7ef43353ea55298973058R178)

Attached is a small bash script + readme file explaining how to reproduce the issue
[tuist-parallel-dump-failure.zip](https://github.com/user-attachments/files/18041583/tuist-parallel-dump-failure.zip)

### How do we reproduce it?

Follow the steps in the readme file, but basically 
1) Download the attached zip
2) unzip
3) cd into the unzipped folder
4) Make sure you are using Tuist 4.33.0
5) Run `chmod +x run.sh`
6) Run the script `./run.sh`

### Error log

Some of the output files would show something like
```sh
We received an error that we couldn't handle:
    - Localized description: The operation couldn’t be completed. (_NIOFileSystem.FileSystemError error 1.)
```

### macOS version

14.7.1

### Tuist version

4.33.0

### Xcode version

15.4 (also with 16 and 16.1)

**Comment by fortmarek** [comment]
Hey @danibachar 👋 

I am unable to reproduce the error even when I ran the command multiple times:
```
You are using tuist version 4.33.0
The issue will happen from tuist version 4.33.0
Cloning Tuist repo to use fixtures
Cloning into 'tuist'...
remote: Enumerating objects: 4161, done.
remote: Counting objects: 100% (4161/4161), done.
remote: Compressing objects: 100% (2993/2993), done.
remote: Total 4161 (delta 960), reused 2528 (delta 647), pack-reused 0 (from 0)
Receiving objects: 100% (4161/4161), 12.06 MiB | 15.52 MiB/s, done.
Resolving deltas: 100% (960/960), done.
Cleaning Tuist artifcats...
Successfully cleaned artifacts at path /Users/marekfort/.cache/tuist/Manifests
Successfully cleaned artifacts at path /Users/marekfort/.cache/tuist/ProjectDescriptionHelpers
Successfully cleaned artifacts at path /Users/marekfort/.cache/tuist/EditProjects
Successfully cleaned artifacts at path /Users/marekfort/.cache/tuist/Runs
Dumping in prallel, expect xcodebuild/NIO swift/file system error
Done dumping!
```

Would you be able to update the reproducer, so we can see the issue happening?

> If indeed that change is in charge of it - would the Tuist team be open to revert it given we have the protection check [here](https://github.com/tuist/tuist/commit/affac241587d77c64d87550e6031fa55633efd62#diff-70cba27f5cddb0850785b77ded2a515d37cf0b771fe7ef43353ea55298973058R178)

`CachedManifestLoader` is caching at the manifest level whereas that commit added caching at the helpers level. Ideally, we'd keep both.

**Comment by danibachar** [comment]
@fortmarek - did you look into the `.txt` files that are created? Those should include the error - not the script itself, as I pipe stdout and stderr to the file

**Comment by fortmarek** [comment]
Sorry, I missed that.

`tuist dump` is not something I'd expect users to run in parallel – other commands like `tuist generate` (or any other command using the `CachedManifestLoader`) would have the same issue.

Since we use the disk for caching, it will always be tricky to ensure everything works well when you have multiple processes writing into the same place.

What we can do here is to catch the error and silently allow it as we don't care the file already exists in this instance as they should have the same content since they're uniquely identified by the hash.

**Comment by danibachar** [comment]
Yea that makes sense, this will be a simple try/catch around [this function](https://github.com/tuist/tuist/blob/f5a97e6eb3ed7be7bb7c988ed88b98888b454316/Sources/TuistLoader/Loaders/CachedManifestLoader.swift#L163C19-L163C32) correct? Or the `write` call in the downstream function?

**Comment by fortmarek** [comment]
> Yea that makes sense, this will be a simple try/catch around [this function](https://github.com/tuist/tuist/blob/f5a97e6eb3ed7be7bb7c988ed88b98888b454316/Sources/TuistLoader/Loaders/CachedManifestLoader.swift#L163C19-L163C32) correct?

Yup, although, I'd try to make the `try-catch` lower, ideally, directly around the `fileSystem` call that fails when `tuist dump` commands are run in parallel.
